src/proxy.js

```diff
@@ -244,9 +244,17 @@ export default class Proxy extends EventEmitter {
         }, fromServer => {
           toClient.writeHead(fromServer.statusCode, fromServer.headers)
           fromServer.pipe(toClient)
+          toServer.end();
```

Comment:
Has this been tested with a POST (or anything with a request body)? I'm worried this would prematurely end the flow of data in those cases. If this is for error handling, would it make more sense to move it into the `onError` handler?

Reply:
Thinking more about this, by this point we've received a response from upstream, so we'd definitely expect the request to be finished. Still, doesn't the `pipe()` call below do this automatically?

Reply:
There's nothing in the docs that suggests that `pipe` will do that: https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_readable_pipe_destination_options

This missing `end` might well be an issue for some socket timeout errors we saw in james.